src/Components/Components/src/Auth/AuthorizeRouteView.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Components.Auth;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Combines the behaviors of <see cref="AuthorizeView"/> and <see cref="RouteView"/>,
+    /// so that it displays the page matching the specified route but only if the user
+    /// is authorized to see it.
+    ///
+    /// Additionally, this component supplies a cascading parameter of type <see cref="Task{AuthenticationState}"/>,
+    /// which makes the user's current authentication state available to descendants.
+    /// </summary>
+    public sealed class AuthorizeRouteView : RouteView
```

Comment:
Sealed just because I can't anticipate any useful scenario for subclassing this one in particular.